src/Akka.Persistence.Sql.Linq2Db/Journal/DAO/BaseByteArrayJournalDao.cs

```diff
@@ -436,9 +480,9 @@ await db.GetTable<JournalRow>()
         {
             using (var db = _connectionFactory.GetConnection())
             {
-                return await MaxSeqNumberForPersistenceIdQuery(db,
+                return (await MaxSeqNumberForPersistenceIdQuery(db,
                     persistenceId,
-                    fromSequenceNr).FirstOrDefaultAsync();
+                    fromSequenceNr).MaxAsync()).GetValueOrDefault(0);
```

Comment:
Linq2Db Expects that `.MaxAsync()` on `long` or `int` types will return a value; otherwise an exception is thrown. So we do this instead on the `IQueryable<long?>`.